core-java/src/main/java/com/baeldung/system/DetectOS.java

```diff
@@ -0,0 +1,50 @@
+package com.baeldung.system;
+
+import java.util.Locale;
+
+import org.apache.commons.lang3.SystemUtils;
+
+public class DetectOS {
+
+    private static final DetectOS INSTANCE = new DetectOS();
+    private static OperatingSystem detectedOS;
```

Comment:
Let's simplify the design and not store the result anywhere - let's simply return the result from the method